src/data_point_routines.f90

```diff
@@ -404,11 +404,16 @@ SUBROUTINE DATA_POINTS_DESTROY(DATA_POINTS,ERR,ERROR,*)
     ENTERS("DATA_POINTS_DESTROY",ERR,ERROR,*999)
 
     IF(ASSOCIATED(DATA_POINTS)) THEN
-     CALL DATA_POINTS_FINALISE(DATA_POINTS,ERR,ERROR,*999)
+      IF (ASSOCIATED(DATA_POINTS%REGION)) THEN
+        NULLIFY(DATA_POINTS%REGION%DATA_POINTS)
+      ELSE
+        CALL FLAG_ERROR("Data_points region is not associated.",ERR,ERROR,*999)
```

Comment:
There probably shouldn't be an error here as we will never be able to destroy the data points. If the region is not associated then just finalise the data points.